addons/pos_restaurant/static/src/app/pos_store.js

```diff
@@ -26,6 +26,86 @@ patch(PosStore.prototype, "pos_restaurant.PosStore", {
         if (this.globalState.config.module_pos_restaurant) {
             this.setActivityListeners();
             this.showScreen("FloorScreen", { floor: this.globalState.table?.floor || null });
+            this.initTableOrderCount();
+        }
+    },
+
+    async initTableOrderCount() {
+        const result = await this.orm.call("pos.config", "get_tables_order_count", [
+            this.globalState.config.id,
+        ]);
+        this.ws_syncTableCount(result);
+    },
+
+    // Will receive all messages coming from the server via
+    // the websocket and dispatch them to the different utility functions.
+    handleBusMessages(detail) {
+        this._super(...arguments);
+
+        if (detail.type === "table_order_count") {
+            this.ws_syncTableCount(detail.payload);
+        } else if (detail.type === "table_changed") {
+            this.ws_syncTableChanges(detail.payload);
+        } else if (detail.type === "disable_floor") {
+            this.ws_disableFloor(detail.payload);
+        }
+    },
+
+    // Handles server bus messages of type `disable_floor`,
+    // it will delete the floor and its tables.
+    ws_disableFloor(data) {
```

Comment:
I'm thinking of extracting these `bus message handlers` to their own service. This way, we don't pollute the pos_store with these handlers.

Reply:
We need to talk about this, there are some technical impossibilities to do this. We can perhaps create a new class instantiated in the pos_store by passing it globalState etc.

Reply:
Maybe you can do something like:

```js
export class BusHandlerService {
  static serviceDependencies = ['pos', 'orm', 'bus'];
  constructor(...args) {
    this.setup(...args);
  }
  setup({ pos, orm, bus }) {
    this.pos = pos;
    this.orm = orm;
    this.bus = bus; // if needed
  }
  dispatch(message) {
    <switch statement here for delagating the message handling>
  }
}

registry.category("services").add({
  deps: BusHandlerService.serviceDependencies,
  start: async(env, deps) => {
    const { pos, bus } = deps;
    const busHandler = new BusHandlerService(deps);

    // initialize bus_service and listen on pos_config-`id` channel
    bus.addChannel(`pos_config-${pos.globalState.config.id}`);
    bus.addEventListener("notification", async (message) =>
            message.detail.map((detail) => {
                this.busHandler.dispatch(detail);
            })
        );
    
    await busHandler.initTableOrderCount();
    
    return busHandler;
  }
})
```